datafusion/src/physical_plan/coercion_rule/binary_rule.rs

```diff
@@ -162,12 +162,141 @@ fn get_comparison_common_decimal_type(
     }
 }
 
+// Convert the numeric data type to the decimal data type.
+// Now, we just support the signed integer type and floating-point type.
+fn convert_numeric_type_to_decimal(numeric_type: &DataType) -> Option<DataType> {
+    match numeric_type {
+        DataType::Int8 => Some(DataType::Decimal(3, 0)),
+        DataType::Int16 => Some(DataType::Decimal(5, 0)),
+        DataType::Int32 => Some(DataType::Decimal(10, 0)),
+        DataType::Int64 => Some(DataType::Decimal(20, 0)),
+        // TODO if we convert the floating-point data to the decimal type, it maybe overflow.
+        DataType::Float32 => Some(DataType::Decimal(14, 7)),
+        DataType::Float64 => Some(DataType::Decimal(30, 15)),
+        _ => None,
+    }
+}
+
+fn mathematics_numerical_coercion(
```

Comment:
This function looks almost the same as `numerical_coercion`  -- which is used for equality

What is the reason for making a new function rather than extending `numerical_coercion` with Decimal support?

Reply:
![image](https://user-images.githubusercontent.com/7450163/149791191-ed11a0f5-8794-44b7-a695-b83b82e9fb3c.png)
The `numerical_coercion` is used by other coercion functions:
1. `numerical_coercion`->`eq_coercion`  -> `Operator::IsDistinctFrom | Operator::IsNotDistinctFrom => `
2. `numerical_coercion` -> `dictionary_value_coercion`
If I extend the `numerical_coercion`, it may take a side effect to other coercion logical.
I will refactor and merge these two together and make it clear if there is no conflict.
@alamb

Reply:
thank you

Reply:
did you try extending `numerical_coercion` (and if so, did it have any side effects)?

Reply:
@alamb  I will extend it in the follow-up pull request and support dict and `distinct` with decimal.
In the follow-up pull request, I think we can refactor out a common part.